src/postgres-util/src/lib.rs

```diff
@@ -89,8 +90,67 @@ pub fn make_tls(config: &PostgresConfig) -> Result<MakeTlsConnector, anyhow::Err
     Ok(tls_connector)
 }
 
-/// Fetches table schema information from an upstream Postgres source for all tables that are part
-/// of a publication, given a connection string and the publication name.
+/// Gets the OID of the named type.
+///
+/// The second argument names the type to look up, and is of the form
+/// `(<schema>?, <item>)`.
+///
+/// # Errors
+///
+/// - If the supplied name does not unambiguously name a type in the upstream
+///   Postgres database's catalog.
+pub async fn get_typname_oid(
+    config: &Config,
+    (schema, item): (Option<&str>, &str),
+) -> Result<u32, anyhow::Error> {
```

Comment:
We have an `Oid` data type defined in `mz_repr::adt::Oid`, should we use that instead here?